### PR TITLE
Add mobile web app support

### PR DESCRIPTION
**Pull request by tris**
Tested on:
- Safari Technology Preview Release 159 on macOS Ventura 13.1
- Google Chrome 109.0.5414.119 on macOS Ventura 13.1
- Safari on iOS 16.2 on iPhone 14 Pro Max via iOS Simulator
- Safari on iOS 16.1 on iPhone 14 Pro Max

Open issues:
- The bat doesn't seem to be centered in `mstile-150x150.png`.  I'm not sure why... that's the way it came out of realfavicongenerator.net.  Still need to test on Windows.

Notes:
- I needed a higher-res version of the logo for some of this stuff.  I found the SVG (https://www.flaticon.com/free-icon/bat_1205671) but it needed some extra work in Inkscape to make it true to the logo.  I removed most of the black contour lines, while the remaining two formed a rib at the midway point on the wings, which I recolored to #4e5173 and then used Path > Division to slice along the sharp bottom contour of the wing.
- The Inkscape and Pixelmator assets (`bat-icon.svg`, `bat-icon.pxd`) perhaps belong somewhere other than `.../public/static/img/`.  They aren't needed by the browser but will be needed if any work needs to be done with the logo again.
- https://realfavicongenerator.net served as a starting point to generate the rest of the image assets and boilerplate.
- For iOS, the `apple-touch-icon` isn't supposed to have a transparent background.  (I tried it, and it's glitchy -- transparent during animations and then transitioning to an ugly flat black background.)  I used Inkscape and Pixelmator to make a proper iOS-style icon with a subtle gradient, shrinking the logo to 93.75% to fit within the design template. <br>image
- To properly support Safari pinned tabs and particularly the MacBook Touch Bar, a `mask-icon` is required.  I found three problems with that: (1) the mask-icon must be monochrome, (2) the mask-icon appears to override the favicon even in unpinned tabs, and (3) the color is independent of light/dark mode, so a neutral color must be picked.  The color I picked (#5bbad5) doesn't look great, so I'm skipping the mask-icon, and the Touch Bar can just bite me.
- For Windows Metro, I chose #da532c for the tile background (Dark Orange -- one of the 9 suggested Metro colors).  I'm not in love with it.
- `service-worker.js` must be in the root directory, *or* the file must be served with an extra header (`Service-Worker-Allowed: /`).  The former option seemed less brittle.
- The service worker isn't required for iOS, but is required for Chrome.  (It works as an installable app on desktop Chrome!)
- Lighthouse was invaluable for debugging the PWA.
- A real SSL certificate (not self-signed) is required for the PWA to work fully (Chrome on OS X was particularly difficult here).  (Also, in Safari, HTML5 Geolocation won't work over plain HTTP.)

Bugs and future work:
- When the site is inaccessible, iOS seems to get into a reload loop.  Not sure why that is.  Once deployed to production (with legit SSL) it will hopefully use the service worker and serve up the offline page.
- The Ozone logo is partially obscured by the "home indicator" at the bottom of the screen.  ([Real apps can hide this.]( https://developer.apple.com/documentation/uikit/uiviewcontroller/2887510-prefershomeindicatorautohidden))
- [Deal with the notch](https://stackoverflow.com/questions/60154236/is-viewport-fit-cover-no-longer-working-on-the-ios-safari).  (For now, all content is below the notch.)
- Add more icon sizes to `manifest.json` (up to 1024x1024)
- Set `background_color` and `theme_color` more appropriately
- Make a proper set of `apple-touch-startup-image`s -- I made a square 2048x2048 one (`apple-touch-startup-image.png`) based on the current splash screen (via screen recording), but it seems there needs to be an image for each screen size (https://stackoverflow.com/a/57338818).
- Try using the service worker to actually cache stuff, or even update tracks in the background (may be useful in poor signal areas).
- Improve the offline page (at least add a reload button?)
- Add a [`maskable` icon](https://w3c.github.io/manifest/#icon-masks) (Lighthouse throws a warning about this)
- Add [`shortcuts`](https://developer.mozilla.org/en-US/docs/Web/Manifest/shortcuts) (see YouTube's manifest as an example)
- Add [`share_target`](https://developer.mozilla.org/en-US/docs/Web/Manifest/share_target) (to visualize an igc or task, perhaps?)
- Add [`screenshots`](https://developer.mozilla.org/en-US/docs/Web/Manifest/screenshots) (?) and [submit to Google Play Store](https://developers.google.com/codelabs/pwa-in-play)

More resources:
- [Use Progressive Web Apps (Google Chrome)](https://support.google.com/chrome/answer/9658361)
- [Learn PWA (web.dev)](https://web.dev/learn/pwa/)
- [Configuring Web Applications (Safari Web Content Guide)](https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariWebContent/ConfiguringWebApplications/ConfiguringWebApplications.html)
- [Supported Meta Tags (Safari HTML Reference)](https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariHTMLRef/Articles/MetaTags.html)
- [iOS App Icon Production Templates (Apple Design Resources)](https://developer.apple.com/design/resources/#ios-apps)
- [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk)
- https://www.youtube.com/manifest.webmanifest
- [How to Favicon in 2021](https://www.leereamsnyder.com/blog/favicons-in-2021)
- [Are you using SVG favicons yet?  A guide for modern browsers.](https://medium.com/swlh/are-you-using-svg-favicons-yet-a-guide-for-modern-browsers-836a6aace3df)

### CODE REVIEW

File: apps/fxc-front/public/static/archives.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
   <meta charset="utf-8" />
```
Comment (vicb) [inline apps/fxc-front/public/static/archives.html:12]: Plese delete this file. It's not used any more.
I left it around to use when porting to the new arch but should have deleted it as part of https://github.com/vicb/flyxc/commit/4a1aef2f3acf8dc5984b64885df34ac8ffdd2562

File: apps/fxc-front/public/static/manifest.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "icons": [
+        {
+            "src": "/static/img/android-chrome-192x192.png",
+            "sizes": "192x192",
+            "type": "image/png"
+        }
+    ],
+    "theme_color": "#ffffff",
+    "background_color": "#ffffff",
+    "start_url": "https://flyxc.app/?source=pwa",
```
Comment (vicb) [inline apps/fxc-front/public/static/manifest.json:13]: I don't think we need `?source=pwa` ?

**Review by vicb — COMMENTED** [review]
Thanks for your PR.

Please add your name at https://github.com/vicb/flyxc/blob/master/README.md#contributors and https://github.com/vicb/flyxc/blob/2d6d70c0432a14c17bf4fafa967a2a8ed6a29fe9/apps/fxc-front/src/app/components/ui/about-modal.ts#L18-L22

File: apps/fxc-front/public/static/manifest.json (hunk 2 above)
Comment (tris) [reply apps/fxc-front/public/static/manifest.json]: I cargo culted that from web.dev, but definitely unnecessary (except for analytics etc).  Removed now.

**Comment by vicb** [comment]
Thanks for the change and the great description.
I'll come to the review as soon as I have some time.

File: apps/fxc-front/public/static/offline.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  <head>
+    <title>Offline</title>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="theme-color" content="#3f51b5">
+    <link rel="manifest" href="/static/manifest.json">
+    <style>
+      body {
+        font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
```
Comment (vicb) [inline apps/fxc-front/public/static/offline.html:12]: We probably don't want apple specific or exotic fonts in here.
We should re-use whatever flyxc.css uses.

File: apps/fxc-front/public/static/offline.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
...2hmNJDagGEMgBClA5WLLoETp1ck1HLDj/TfCYMNP28LsfGHyOORKDeZiaM7xrZ2nIIzTVOGuW1HTBravGxpURiDYymeq9FLu1kPZuf7XIei9+rdJTJ3l5Edf3K31FEzLM5YKGaolMd8FAGJ5kwQDwwUOS4Qfn56pQACFFeaP0oN4K5kwkD1cQ/fgoI4xiP3b8yYbxoVrxCAaFjwNFRAGnT+OA9T3drKiBofaRpOEfL4TQyNlDYL+axuIXHZLVS9ieywYz91yxW3h9RuGPA6GbFzwVFjEtkOxYnApAYUOqIZSm9phtOQ1aRZ8JYz8/4F3y8bpLA8SwD46s8doPO4Gtbv909karjEJd+/j4aG6NmFfQ+Y+XSAZ16TPBdKO2/y+Oxkcfj3C6BAiPEv/HzP8jPf9RwjG5rXOVwCcq4dshsWwsYkwmQHLDkgCAFhBhgYgaFphK7EHpxI798ivcPdfF8IO6neby+zsTxCxoffk3GlNl2uIQGigUIOabIpuEi5HCQRoBzNB3xFht6b2/k/QjP+/qKo5zDr63A3A3xM6Ob1xbe/5Of/QnHdEvkx1DlgiWXS1DGNSYEilMJEEqIWzHOQTWAQzVA1XQ4CrT4C/mZb6qAsob3ywPPIXFez/AOE+4hs5qT4xsiBYzCAUknAg5ybPUWJF7Yf59ZECz3sQQ1s/rvK2ikTzmsX+AoK3lczqeRMBE861beX6mscAO8/4RGKpR0KJ31l9IF6ijSRYLwT0kZ0lMJkFywUCY3yBHJKMBBKKSjKGLHGMDiA7AghH477ycDxNcMPFPhOMQ6jm7SCYgjXiGDUDalvZcQ54gF/bUVB2pVpuELKrAgA/A3vB9SVrhOwF/TqSHq5BB6rt5wymvzni6AeIAoqPuEKpogp2k6x6HkLG+F9v7LEgs5+kWH4p5jT0EPLQZ9kfv1fFMhE6oFsvzHicR9dxxgT4ZuUGSITKetuvuZ6lEYq9gdWiEaNY9j4OmElHmzKo9zQpKfRx/iAkTjwyqI0k6zlMWvYZ4hlOPfdBYlCw4v/CMUG9VxREnKBEEdH0SKM5zWniG91ECniJg6Y7VXcwHinfMCKPscYLQzOVpIPLBZcymzZJ1q+A0HDN5xSuG1ol8n4NeYLGKfOh2merRf9ZQYwOnt//fWnB6C6W16C2//J8AA8rWjFYGIlNoAAAAASUVORK5CYII=">
```
Comment (vicb) [inline apps/fxc-front/public/static/offline.html:26]: Can we use an external image here and cache it at install time ?
Would be easier to inspect that base64

File: apps/fxc-front/service-worker.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const CACHE_NAME = 'offline';
+const OFFLINE_URL = '/static/offline.html';
+
+self.addEventListener('install', function(event) {
+  console.log('[ServiceWorker] Install');
```
Comment (vicb) [inline apps/fxc-front/service-worker.js:5]: remove

**Review by vicb — COMMENTED** [review]
It would be awesome to support installing flyxc as a web app.
I don't know much about so I need to read more before I can fully review.
Hopefully I can make some progress this week.

What do you think of using [this icon](https://www.flaticon.com/free-icon/aviator_2636637)

File: apps/fxc-front/public/static/offline.html (hunk 3 above)
Comment (elgandoz) [reply apps/fxc-front/public/static/offline.html:12]: https://css-tricks.com/snippets/css/system-font-stack/

**Pull request closed without merging**